packages/vscode-extension/src/handlers.ts

```diff
@@ -105,10 +105,27 @@ export async function activate(): Promise<Result<Void, FxError>> {
   try {
     syncFeatureFlags();
 
-    if (isValidProject(getWorkspacePath())) {
+    const validProject = isValidProject(getWorkspacePath());
+    if (validProject) {
       ExtTelemetry.sendTelemetryEvent(TelemetryEvent.OpenTeamsApp, {});
     }
 
+    const expService = exp.getExpService();
+    if (expService) {
+      if (
+        !validProject &&
+        (await expService.getTreatmentVariableAsync(
+          TreatmentVariables.VSCodeConfig,
+          TreatmentVariables.SidebarWelcome,
+          true
+        ))
+      ) {
```

Comment:
the if condition looks strange. should we check !validProject first, then getTreatmentVariableAsync().

Reply:
I think the getTreatmentVariableAsync will not be executed if !validProject returns false